File: app/src/main/java/com/umbrella/budgetapp/Database/Collections/Currency.java
package com.umbrella.budgetapp.Database.Collections;

public class Currency {
    public static final String COLLECTION = "currency";
    public static final String COUNTRY = "countryID";
    public static final String RATE = "usedRate";
    public static final String POSITION = "position";

    private String CountryID;
    private String UsedRate;
    private int Position;

    public Currency() {
        //Empty constructor needed.
    }

    public Currency(String countryID, String usedRate, int position) {
        CountryID = countryID;
        UsedRate = usedRate;
        Position = position;
    }

    public String getCountryID() {
        return CountryID;
    }

    public String getUsedRate() {
        return UsedRate;
    }

    public int getPosition() {
        return Position;
    }

    public void setCountryID(String countryID) {
        CountryID = countryID;
    }

    public void setUsedRate(String usedRate) {
        UsedRate = usedRate;
    }

    public void setPosition(int position) {
        Position = position;
    }
}

File: app/src/main/java/com/umbrella/budgetapp/database/collections/Currency.kt
package com.umbrella.budgetapp.database.collections

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.android.parcel.Parcelize
import java.math.BigDecimal

@Parcelize
@Entity(tableName = "currencies")
data class Currency(
        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = "currency_id")
        val id: Long? = null,

        @ColumnInfo(name = "currency_country_ref")
        val countryRef: Long? = 0,

        @ColumnInfo(name = "currency_used_rate")
        var usedRate: BigDecimal? = BigDecimal.ONE,

        @ColumnInfo(name = "currency_position")
        var position: Int? = 0
) : Parcelable
